MapboxVision/Services/VideoSource/ObservableVideoSource.swift

```diff
@@ -4,38 +4,97 @@ import Foundation
  Helper class handling observers: storing, releasing, notifying.
  Observers are held weakly by the instance of the class.
  You may inherit your video source from this class to avoid handling observers yourself.
+
+ - Warning:
+ The implementation uses a recursive lock, thus you must not call `add(observer:)` or `remove(observer:)` methods from `notify` closure.
```

Comment:
Wait, it's a non-recursive lock, isn't it?

Reply:
LOL, yes. Thanks.